### DISCUSSION TITLE
Create a user from the dashboard.

### DISCUSSION THREAD

**Discussion by ricardosikic** [supabase/supabase#22067] [Questions] [answered]
Hello, I have a dashboard in my app through which I can add new users to the app. I understand that the signUp function should be used so that they go to auth.users and trigger the configured triggers. But when I test the functionality, it automatically logs me in to the new account I created from the superadmin, so it logs me out of superadmin, so to speak, and shows me the home for the created user. The code I used is as follows:

```
const createNewUserMunicipality = async(name, email, muniId) => {

    try {
      const { user, session, error } = await supabase.auth.signUp(
        {
          email: email,
          password: '12345678',
        },
        {
          data: {
            username: name
          }
        }
      )
      setCreatedUserId(user.id);
      createRelationOnUserRolesTable(user.id, muniId);
    } catch (err) {
      console.log('There was an error creating user municipality', err);
    }
  }
```
Is this the correct way to add new users from the app's administrator dashboard?

**Comment by GaryAustin1** [comment, marked as answer]
You should look at admin.createUser https://supabase.com/docs/reference/javascript/auth-admin-createuser .

Note it has to be run server side as it needs service_role key.

**Reply by ricardosikic** [reply]
Thanks @GaryAustin1 wow its only for v2?

**Reply by GaryAustin1** [reply]
It was in version 1 also.  https://supabase.com/docs/reference/javascript/v1/auth-admin-createuser .  Looks like the docs for v1 are not working well anymore.   You should not be using v1 though as it was deprecated almost a year and 1/2 ago.

**Reply by ricardosikic** [reply]
thanks, yep i guess i have to update my version to v2.

**Reply by mi-dexigner** [reply]
@ricardosikic Hi Rick, I'm facing the same issue. I'm trying to create a new user for the dashboard and retrieve all user data onto my dashboard, but it's not working. When I run the code, it returns an empty array. Here's my code:
try {
  const { data: { users }, error } = await supabase.auth.admin.listUsers();
  console.log('Data:', users);
} catch (error) {
  // Catch any unexpected errors during the process
  console.error('Unexpected error:', error.message);
}
Could you please help me correct it?

**Reply by ricardosikic** [reply]
Have you used service_role_key to use that `create-user` function?

**Comment by mi-dexigner** [comment]
@ricardosikic  yes its working thanks because now i am using the service_role_key instead of the anon and one more question when i deploy the vercel code i found the error "This Edge Function has crashed." but on my local machine everything is fine can you help me
